vita3k/app/src/app_init.cpp

```diff
@@ -101,11 +102,93 @@ void update_viewport(EmuEnvState &state) {
+    else if (env_home != NULL)
+        root_paths.set_config_path(fs::path(env_home) / ".config" / app_name / dir_sep);
+
+    env_data = getenv("XDG_CACHE_HOME");
+    if (env_data != NULL) {
+        root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(env_data)) / app_name / dir_sep);
+        root_paths.set_log_path(fs::path(string_utils::utf_to_wide(env_data)) / app_name / dir_sep);
+    } else if (env_home != NULL) {
+        root_paths.set_cache_path(fs::path(env_home) / ".cache" / app_name / dir_sep);
+        root_paths.set_log_path(fs::path(env_home) / ".cache" / app_name / dir_sep);
+    }
+
+    // Don't assume that base_path is portable.
+    if (fs::exists(root_paths.get_base_path() / "data") && fs::exists(root_paths.get_base_path() / "lang") && fs::exists(root_paths.get_base_path() / "shaders-builtin"))
+        root_paths.set_shared_path(root_paths.get_base_path());
+    else if (env_home != NULL)
+        root_paths.set_shared_path(fs::path(env_home) / ".local/share" / app_name / dir_sep);
+
+    env_data = getenv("XDG_DATA_DIRS");
+    if (env_data != NULL) {
+        auto env_paths = string_utils::split_string(env_data, ':');
+        for (std::vector<std::string>::iterator iter = env_paths.begin(); iter != env_paths.end(); iter++) {
+            if (fs::exists(fs::path(*iter) / app_name)) {
+                root_paths.set_shared_path(fs::path(string_utils::utf_to_wide(*iter)) / app_name / dir_sep);
+                break;
+            }
+        }
+    } else if (env_home_data != NULL) {
+        if (fs::exists(fs::path(string_utils::utf_to_wide(env_home_data)) / app_name / "data") && fs::exists(fs::path(string_utils::utf_to_wide(env_home_data)) / app_name / "lang") && fs::exists(fs::path(string_utils::utf_to_wide(env_home_data)) / app_name / "shaders-builtin"))
+            root_paths.set_shared_path(fs::path(string_utils::utf_to_wide(env_home_data)) / app_name / dir_sep);
+    }
+#endif
+
+    // Create default preference and cache path for safety
+    if (!fs::exists(root_paths.get_config_path()))
+        fs::create_directories(root_paths.get_config_path());
+
+    if (!fs::exists(root_paths.get_cache_path()))
+        fs::create_directories(root_paths.get_cache_path());
+
+    if (!fs::exists(fs::path(root_paths.get_log_path()) / "shaderlog"))
+        fs::create_directories(fs::path(root_paths.get_log_path()) / "shaderlog");
+
+    if (!fs::exists(fs::path(root_paths.get_log_path()) / "texturelog"))
+        fs::create_directories(fs::path(root_paths.get_log_path()) / "texturelog");
+
+    LOG_INFO("Base path: {}", root_paths.get_base_path_string());
+    LOG_INFO("Shared assets path: {}", root_paths.get_shared_path_string());
+    LOG_INFO("Log path: {}", root_paths.get_log_path_string());
+    LOG_INFO("User config path: {}", root_paths.get_config_path_string());
+    LOG_INFO("User pref path: {}", root_paths.get_pref_path_string());
+    LOG_INFO("User cache path: {}", root_paths.get_cache_path_string());
```

Comment:
Logging is not initialized at this point so it's not showing up great in the console (and won't be logged in the log file). I think either this can get remove or put after the logging::init call, not sure how important it is to have this in the log file.

Reply:
Fixed.